### PR TITLE
Improve get-text behavior for nested blocks

### PR DESCRIPTION
**Pull request by oyeanuj**
Hi @ianstormtaylor, a small PR that improves on the behavior in the case of nested blocks. Currently, the blocks get concatenated joining the last word from the previous block with the first word from the next block.

This PR follows plain serializer tradition behavior by adding a newline when concatenating nested blocks.

### CODE REVIEW

File: src/models/node.js [hunk 1]
```diff
@@ -1179,7 +1179,7 @@ const Node = {
 
   getText() {
     return this.nodes.reduce((string, node) => {
-      return string + node.text
+      return node.kind == 'block' ? `${string}\n${node.text}` : string + node.text
```
Comment (YurkaninRyan) [inline src/models/node.js]: Even if you have only one block, it will always start with a link break.  I don't think we want that right?

Comment (oyeanuj) [inline src/models/node.js]: Just about to push in a fix for this (also, why the tests are failing) :)

Comment (YurkaninRyan) [inline src/models/node.js]: Nice :D good idea btw i'm all for this pr

**Comment by oyeanuj** [comment]
So, the other two errors are stemming from a mismatch between `block.text.length` and `getOffset` for a block. It shows up on `deleteAtRange` transform when the [`offset` is created](https://github.com/ianstormtaylor/slate/blob/master/src/transforms/at-range.js#L87), and (then via `splitNodeByKey` method) fails at [this line of code](https://github.com/ianstormtaylor/slate/blob/master/src/models/node.js#L1196). Earlier both would be the same here but now `this.length` accounts for the newline and `offset` doesn't.

@ianstormtaylor Question for you since you know this code the best, should `getOffset` be modified [here](https://github.com/ianstormtaylor/slate/blob/master/src/models/node.js#L1045) as well to account for the newline change in this PR above?

**Comment by oyeanuj** [comment]
I am now thinking that a less messy option might be to modify [the plain serializer code here](https://github.com/ianstormtaylor/slate/blob/master/src/serializers/plain.js#L50) to compute text within a block with newlines instead of modifying `getText` (like above). So, I'd probably close this PR and issue a new one with changes in the plain serializer.

The only downside seems to be maybe performance impact(?). Any thoughts @YurkaninRyan @ianstormtaylor?

**Comment by ianstormtaylor** [comment]
@oyeanuj I think your intuition is right about making this change in the `Plain` serializer instead of in the `node.text`, because that text is more for internal offset calculation type things. (Although I do understand that it's a bit confusing in the `document.text` case.)

**Comment by oyeanuj** [comment]
Great, then I'll close this PR and create a new one with that change!

**Pull request closed without merging**